### ISSUE TITLE
oracle oci8 support mac arm64?,for example mac m1

### ISSUE THREAD

**Issue by PengFei-io** [mattn/go-oci8#435]
oracle oci8 support mac arm64?,for example mac m1.
when use the mac m1 computer run the oci8 drive. its will occurred error.not found oci8.

**Comment by cjbj** [comment]
Until we (Oracle) release it for Apple Silicon, you'll need to use Instant Client for Intel via Rosetta, or use the Linux ARM Instant Client in a Docker container.

**Comment by PengFei-io** [comment]
all right,thanks

**Comment by fairclothjm** [comment]
It looks like Oracle has published the instant client for Apple Silicon: https://docs.oracle.com/en/database/oracle/oracle-database/23/mxcli/overview-of-oracle-database-client-installation.html

**Comment by cjbj** [comment]
Yes. Also see https://medium.com/oracledevs/installing-oracle-instant-client-for-apple-macos-arm64-m1-m2-m3-2c81f246feb9

**Comment by mattn** [comment]
Yes. Oracle now ships a native macOS ARM64 (Apple Silicon) Instant Client (23.3 and later): https://www.oracle.com/database/technologies/instant-client/macos-arm64-downloads.html — install it, point your oci8.pc (or CGO_CFLAGS/CGO_LDFLAGS with the noPkgConfig build tag) at it, and go-oci8 builds and runs natively on darwin/arm64. At the time this issue was opened no ARM64 client existed, which was the actual blocker.
